database/database.go

```diff
@@ -193,75 +189,18 @@ func (db *DB) UserDelete(ctx context.Context, u *User) error {
 	return nil
 }
 
-// UserUpdate saves the user in the DB.
+// UserUpdate changes the user's data in the DB.
+// It never changes the id or sub of the user.
 func (db *DB) UserUpdate(ctx context.Context, u *User) error {
-	if !u.Email.Validate() {
-		return ErrInvalidEmail
-	}
-	// Check for an existing user with this email.
-	users, err := db.managedUsersByField(ctx, "email", string(u.Email))
-	if err != nil && !errors.Contains(err, ErrUserNotFound) {
-		return errors.AddContext(err, "failed to query DB")
-	}
-	// Sanity check.
-	if len(users) > 1 {
-		build.Critical("More than one user found with email", u.Email)
-		return ErrEmailAlreadyUsed
-	}
-	if len(users) > 0 && !bytes.Equal(u.ID[:], users[0].ID[:]) {
-		return ErrEmailAlreadyUsed
-	}
-	// TODO What if we have a race a user gets this email right at this point?
 	// Update the user.
 	filter := bson.M{"_id": u.ID}
 	update := bson.M{"$set": bson.M{
-		"_id":       u.ID,
-		"firstName": u.FirstName,
-		"lastName":  u.LastName,
-		"email":     u.Email,
```

Comment:
why did you drop these fields?

Reply:
These are managed by Kratos. The user can edit their names, change their password and email and get the respective confirmations and checks via Kratos. We also have them in the JWT token, so we don't need to store them in the DB.

Reply:
I think you could argue it would actually be nice to have this information duplicated across both databases. That way we can query everything with mongo, and don't have to resort to writing manual queries to mysql (or cockroach - not sure what we landed on)..

If it is part of the JWT, perhaps we can update the user object on login every time. That's not necessarily a bad thing because we can take advantage of this event and add `lastLoginAt` or something 🤷‍♂️  super useful to have all of this information in mongo imo.

Reply:
I was thinking a lot about this and I don't see a case in which we would need this information and not have a JWT token to get it from. If we have it in Mongo we would need to check with the JWT every time before we use it in order to make sure our copy is up to date. Which renders having it entirely moot.

Put another way, this is an issue of separation of concerns - all user identity information is handled by Kratos while accounts handles only the status and history of the user.